prober/icmp.go

```diff
@@ -249,7 +256,24 @@ func ProbeICMP(ctx context.Context, target string, module config.Module, registr
 	}
 	level.Info(logger).Log("msg", "Waiting for reply packets")
 	for {
-		n, peer, err := socket.ReadFrom(rb)
+		var n int
+		var peer net.Addr
+		var err error
+		var ttl int
+
+		if ip.IP.To4() == nil {
```

Comment:
The main point of having v4Conn is to make v4 and v6 sockets look the same, this code here is completely undoing that polymorphism so we end up with the costs of that without any of the benefits.

Reply:
I added a new version without the v4Conn wrapper class. I would need to forcely morph a ipv4.RawConn into a ipv4.PacketConn. It might be a good reason for go not putting a common public interface for RawConn and PacketConn.

Without v4Conn, it looks cleaner with less LoC.